workitem/workitem_repository.go

```diff
@@ -83,6 +103,119 @@ func (r *GormWorkItemRepository) Delete(ctx context.Context, ID string) error {
 	return nil
 }
 
+// Reorder reorders the given work item in storage. Version must be the same as the one int the stored version
+// returns NotFoundError, VersionConflictError, ConversionError or InternalError
+func (r *GormWorkItemRepository) Reorder(ctx context.Context, wi app.WorkItem) (*app.WorkItem, error) {
+	res := WorkItem{}
+	id, err := strconv.ParseUint(wi.ID, 10, 64)
+	if err != nil || id == 0 {
+		return nil, errors.NewNotFoundError("work item", wi.ID)
+	}
+
+	log.Printf("looking for id %d", id)
+	tx := r.db.First(&res, id)
+	if tx.RecordNotFound() {
+		log.Printf("not found, res=%v", res)
+		return nil, errors.NewNotFoundError("work item", wi.ID)
+	}
+	if tx.Error != nil {
+		return nil, errors.NewInternalError(err.Error())
+	}
+	if res.Version != wi.Version {
+		return nil, errors.NewVersionConflictError("version conflict")
+	}
+
+	wiType, err := r.wir.LoadTypeFromDB(wi.Type)
+	if err != nil {
+		return nil, errors.NewBadParameterError("Type", wi.Type)
+	}
+
+	res.Version = res.Version + 1
+	res.Type = wi.Type
+	res.Fields = Fields{}
+	var order float64
+
+	if wi.Fields[PreviousItem] == nil && wi.Fields[NextItem] == nil {
```

Comment:
PreviousItem & NextItem are not WI fields, the UI doesn't know. The API is Move X, Y before Z